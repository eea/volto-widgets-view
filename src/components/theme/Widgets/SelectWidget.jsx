import React from 'react';
import cx from 'classnames';

export const SelectWidget = ({ children, className }) => (
  <span className={cx(className, 'select', 'widget')}>
    {children?.title || children?.token || children}
  </span>
);
