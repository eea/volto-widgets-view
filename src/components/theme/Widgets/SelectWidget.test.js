import React from 'react';
import renderer from 'react-test-renderer';
import { SelectWidget } from './SelectWidget';

describe('SelectWidget', () => {
  it('renders an empty select view widget component', () => {
    const component = renderer.create(<SelectWidget />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('renders a select view widget component', () => {
    const component = renderer.create(
      <SelectWidget className="metadata">{{ title: 'Foo Bar' }}</SelectWidget>,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('renders a full select view widget component', () => {
    const component = renderer.create(
      <SelectWidget className="metadata">
        {{ title: 'Foo Bar', token: 'foobar' }}
      </SelectWidget>,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
