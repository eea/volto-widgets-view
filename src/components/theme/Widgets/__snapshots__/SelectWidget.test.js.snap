// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectWidget renders a full select view widget component 1`] = `
<span
  className="metadata select widget"
>
  Foo Bar
</span>
`;

exports[`SelectWidget renders a select view widget component 1`] = `
<span
  className="metadata select widget"
>
  Foo Bar
</span>
`;

exports[`SelectWidget renders a simple select view widget component 1`] = `
<span
  className="metadata select widget"
>
  Foo Bar
</span>
`;
